packages/synapse-react-client/src/components/EntityFinder/useEntitySelection.ts

```diff
@@ -24,40 +27,72 @@ export function useEntitySelection(selectMultiple: boolean) {
     [],
   )
 
+  type ReducerAction =
+    | {
+        type: 'toggleSelection'
+        toggledReferences: Reference | Reference[]
+      }
+    | {
+        type: 'setSelection'
+        selection: EntitySelectionMapType
+      }
+
   /**
    * Given the existing selections and a list of toggled references, return the new list of selections
-   * @param selected
-   * @param toggledReference
+   * @param currentState
+   * @param action
    * @returns
    */
   const entitySelectionReducer = useCallback(
     (
-      selected: EntitySelectionMapType,
-      toggledReferences: Reference | Reference[],
+      currentState: EntitySelectionMapType,
+      action: ReducerAction,
     ): EntitySelectionMapType => {
```

Comment:
This reducer previously only allowed toggling one or more references. Change the signature to also allow `setSelection`